src/qt/frame.cpp

```diff
@@ -208,22 +208,23 @@ QScrollArea *wxFrame::QtGetScrollBarsContainer() const
 
 void wxFrame::DoGetClientSize(int *width, int *height) const
 {
-    wxWindow::DoGetClientSize(width, height);
-
-    // Adjust the height, taking the status and menu bars into account, if any:
-    if ( height )
+    QMainWindow *main_window = GetQMainWindow();
+    QSize size;
+    if ( !main_window->isVisible() )
```

Comment:
This is very suspicious: so the client size of the frame will change if you hide/show it back? I don't think it's a good idea to make it behave like this.

Why can't we just always use the main window size, even if the frame is hidden?

And if we really, really can't, we probably need to cache the client size when it's shown and return it from here instead.